File: 0169-majority-element/0169-majority-element.py
"""
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        
        majority = (len(nums) // 2) + 1
            
        occurrences = {}
        
        for i in range(len(nums)):
            if occurrences.get(nums[i]) == None:
                occurrences[nums[i]] = 1
            else:
                occurrences[nums[i]] += 1
            
            if occurrences[nums[i]] == majority:
                return nums[i]

        return None   
"""

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        num, counter = nums[0], 0
        
        for n in nums:
            if n == num:
                counter += 1
            elif counter > 1:
                counter -= 1 
            else:
                num, counter = n, 1
            
        return num      

File: 0169-majority-element/0169-majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        int majorityElement = nums[0];
        int majorityElementCount = 0;
        for (int i = 0; i < nums.Length; i++)
        {
            if (nums[i] == majorityElement)
            {
                majorityElementCount++;
            }
            else
            {
                majorityElementCount--;
            }
            
            if (majorityElementCount == 0)
            {
                majorityElement = nums[i];
                majorityElementCount = 1;
            }
        }
        return majorityElement;
    }
}